presto-geospatial/src/main/java/com/facebook/presto/plugin/geospatial/GeoFunctions.java

```diff
@@ -496,6 +496,18 @@ public static long stNumGeometries(@SqlType(GEOMETRY_TYPE_NAME) Slice input)
         return ((OGCGeometryCollection) geometry).numGeometries();
     }
 
+    @SqlNullable
+    @Description("Returns a geometry that represents the point set union of the Geometries")
+    @ScalarFunction("ST_Union")
+    @SqlType(GEOMETRY_TYPE_NAME)
+    public static Slice stUnion(@SqlType(GEOMETRY_TYPE_NAME) Slice left, @SqlType(GEOMETRY_TYPE_NAME) Slice right)
+    {
+        OGCGeometry leftGeometry = deserialize(left);
+        OGCGeometry rightGeometry = deserialize(right);
+        OGCGeometry ogcGeometry = leftGeometry.union(rightGeometry);
```

Comment:
It returns null when unions GEOMETRYCOLLECTION EMPTY with GEOMETRYCOLLECTION EMPTY

Reply:
@yaoxin226 Xin, see https://github.com/Esri/geometry-api-java/issues/176 and https://github.com/Esri/geometry-api-java/issues/177 for limitations of the library's `union` implementation. See also the implementation of `ST_ConvexHull` in `GeoFunctions.java`.

Until the next release of the library, let's restrict the input of `ST_Union` to  geometries other than geometry collection. Let's also add explicit handing of empty geometries.

```
if (left.isEmpty) return right;
if (right.isEmpty) return left;
```

Reply:
Thanks for your prompt reply.
So this function only supports POINT and LINE_STRING and POLYGON as the input of the function? Support of Multi-geometries also waits?

Reply:
@yaoxin226 It supports multi-geometries, but not geometry-collection.